packages/fxa-shared/subscriptions/types.ts

```diff
@@ -14,7 +19,26 @@ export interface RawMetadata {
 }
 
 // A mapping of OAuth client ids to their corresponding capabilities.
-export type ClientIdCapabilityMap = Record<string, string[]>;
+export type ClientIdCapabilityMap = Record<string, readonly string[]>;
+export const ClientIdCapabilityMap = {
+  merge(
+    a: ClientIdCapabilityMap,
+    b: ClientIdCapabilityMap
+  ): ClientIdCapabilityMap {
+    return Object.entries(b).reduce((acc, [clientId, capabilities]) => {
+      if (!acc[clientId]) {
+        acc[clientId] = cloneDeep(capabilities);
+        return acc;
+      }
+      for (const capability of capabilities) {
+        if (!acc[clientId].includes(capability)) {
+          acc[clientId] = [...acc[clientId], capability];
+        }
```

Comment:
Presumably this is better in some way than using Set to eliminate duplicates?  Its intent is arguably more clear to some readers with `includes`.

Reply:
Yep, a primary focus here was on code readability and maintenance, especially as we have upcoming work in this file that should be easier to perform now.